crm-project/src/main/java/CRM/service/ItemService.java

```diff
@@ -133,4 +150,16 @@ public List<Item> getAllInBoard(long boardId) {
         Board board = Validations.doesIdExists(boardId, boardRepository);
         return itemRepository.findAllByBoard(board);
     }
+
+    public JpaRepository getRepoToUpdateField(Class jpaRepo){
+        if(jpaRepo.getSimpleName().equals(StatusRepository.class.getSimpleName())){
+            return statusRepository;
+        }
+        else if(jpaRepo.getSimpleName().equals(TypeRepository.class.getSimpleName())){
+            return typeRepository;
+        }
+        else{
+            return itemRepository;
+        }
+    }
```

Comment:
this method has to be in the "common" section
Common -> used in various places and does not validate.

If it is only used in ItemService, then it should be a private method INSIDE ItemService and only there.